### PR TITLE
Mule 15379

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataService.java [hunk 1]
```diff
@@ -168,21 +123,19 @@ public void initialise() throws InitialisationException {
...
   }
 
   public MetadataCache getMetadataCache(String id) {
     try {
-      return caches.get(id);
-    } catch (ExecutionException e) {
-      throw new MuleRuntimeException(createStaticMessage("Could not get the cache with id:" + id), e);
+      return cacheManager.getOrCreateCache(id);
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (estebanwasinger) [inline core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataService.java]: is this on purpose?

File: core/src/main/java/org/mule/runtime/core/internal/metadata/cache/DefaultPersistentMetadataCacheManager.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    });
+  }
+
+  @Override
+  public void dispose(String keyHash) {
+    withKeyLock(keyHash, key -> {
+      try {
+        metadataStore.get().remove(key);
+      } catch (ObjectDoesNotExistException e) {
+        LOGGER.debug("No exact match found: " + key);
```
Comment (estebanwasinger) [inline core/src/main/java/org/mule/runtime/core/internal/metadata/cache/DefaultPersistentMetadataCacheManager.java]: I would change it to something more explicit, key `More than one match for key: $key. Disposing all matching values`

File: core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataService.java (hunk 1 above)
Comment (alegmarra) [reply core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataService.java]: of course not 😥

**Comment by alegmarra** [comment]
TODO: review connection order

**Pull request merged**